examples/kv_addr_book/include/kv_addr_book.hpp

```diff
@@ -0,0 +1,163 @@
+#include <eosio/eosio.hpp>
+using namespace std;
+using namespace eosio;
+
+// this structure defines the data stored in the kv_table
+struct person {
+   eosio::name account_name;
+   eosio::non_unique<eosio::name, string> first_name;
+   eosio::non_unique<eosio::name, string> last_name;
+   eosio::non_unique<eosio::name, string, string, string, string> street_city_state_cntry;
+   eosio::non_unique<eosio::name, string> personal_id;
+   std::pair<string, string> country_personal_id;
+
+   eosio::name get_account_name() const {
+      return account_name;
+   }
+   string get_first_name() const {
+      // from the eosio::non_unique tuple we extract the value with key 1, the first name
+      return get<1>(first_name);
+   }
+   string get_last_name() const {
+      // from the eosio::non_unique tuple we extract the value with key 1, the last name
+      return get<1>(last_name);
+   }
+   string get_street() const {
+      // from the eosio::non_unique tuple we extract the value with key 1, the street
+      return get<1>(street_city_state_cntry);
+   }
+   string get_city() const {
+      // from the eosio::non_unique tuple we extract the value with key 2, the city
+      return get<2>(street_city_state_cntry);
+   }
+   string get_state() const {
+      // from the eosio::non_unique tuple we extract the value with key 3, the state
+      return get<3>(street_city_state_cntry);
+   }
+   string get_country() const {
+      // from the eosio::non_unique tuple we extract the value with key 4, the country
+      return get<4>(street_city_state_cntry);
+   }
+   string get_personal_id() const {
+      // from the eosio::non_unique tuple we extract the value with key 1, the personal id
+      return get<1>(personal_id);
+   }
+};
+
+// helper factory to easily build person objects
+struct person_factory {
+   static person get_person(
+      eosio::name account_name,
+      string first_name,
+      string last_name,
+      string street,
+      string city,
+      string state, 
+      string country,
+      string personal_id) {
+         return person {
+            .account_name = account_name,
+            .first_name = {account_name, first_name},
+            .last_name = {account_name, last_name},
+            .street_city_state_cntry = {account_name, street, city, state, country},
+            .personal_id = {account_name, personal_id},
+            .country_personal_id = {country, personal_id}
+         };
+      }
+};
+
+struct address_table : kv_table<person> {
```

Comment:
This should be moved inside your contract class and annotated with `[[eosio::table]]` just like multi_index so that abi generation will find it.